src/Mod/TechDraw/App/DrawProjectSplit.cpp

```diff
@@ -514,25 +426,49 @@ std::vector<TopoDS_Edge> DrawProjectSplit::scrubEdges(std::vector<TopoDS_Edge>&
 //        Base::Console().Message("DPS::scrubEdges(2) - origEdges is empty\n");     //debug
         return std::vector<TopoDS_Edge>();
     }
-    //HLR usually delivers overlapping edges. We need to refine edge overlaps
-    //into non-overlapping pieces
-    std::vector<TopoDS_Edge> noOverlaps;
-    noOverlaps = DrawProjectSplit::removeOverlapEdges(origEdges);
 
-    //HLR algo does not provide all edge intersections.
-    //need to split edges at intersection points.
-    std::vector<TopoDS_Edge> splitEdges;
-    splitEdges = DrawProjectSplit::splitIntersectingEdges(noOverlaps);
+    TopTools_ListOfShape edgeList;
+    for (auto edge : origEdges) {
+        edgeList.Append(edge);
+    }
+
+    BOPAlgo_Builder bopBuilder;
+    bopBuilder.SetArguments(edgeList);
+    bopBuilder.SetFuzzyValue(FUZZYADJUST*EWTOLERANCE);
+    // Allow modifying edges in place, scrubEdges() caller is expected to back them up
+    bopBuilder.SetNonDestructive(Standard_False);
+    // Because we are interested only in edges, we do not need gluing
+    bopBuilder.SetGlue(BOPAlgo_GlueOff);
+    // No solids in the input list
+    bopBuilder.SetCheckInverted(Standard_False);
+    // Use oriented bound boxes
+    bopBuilder.SetUseOBB(Standard_True);
+    bopBuilder.SetRunParallel(Standard_True);
+
+    bopBuilder.Perform();
+    if (bopBuilder.HasErrors()) {
+        Standard_SStream errorStream;
+        bopBuilder.DumpErrors(errorStream);
+        const std::string &errorStr = errorStream.str();
+        Base::Console().Error("DrawProjectSplit::scrubEdges - OCC fuse failed with error(s):\n%s\n", errorStr.c_str());
+        return std::vector<TopoDS_Edge>();
+    }
+
+    if (bopBuilder.HasWarnings()) {
+        Standard_SStream warnStream;
+        bopBuilder.DumpWarnings(warnStream);
+        const std::string &warnStr = warnStream.str();
+        Base::Console().Warning("DrawProjectSplit::scrubEdges - OCC fuse raised warning(s):\n%s\n", warnStr.c_str());
+    }
 
-    //separate any closed edges (ex circle) from the edge pile so as not to confuse
-    //the edge walker later. Closed edges are added back in the caller after
-    //EdgeWalker finds the faces using the open edges
     std::vector<TopoDS_Edge> openEdges;
-    for (auto& edge : splitEdges) {
-        if (BRep_Tool::IsClosed(edge)) {
-            closedEdges.push_back(edge);
-        } else {
-            openEdges.push_back(edge);
+    const TopoDS_Shape &bopResult = bopBuilder.Shape();
+    if (!bopResult.IsNull()) {
+        TopExp_Explorer explorer(bopResult, TopAbs_EDGE);
+        while (explorer.More()) {
+            const TopoDS_Edge &edge = TopoDS::Edge(explorer.Current());
+            (BRep_Tool::IsClosed(edge) ? closedEdges : openEdges).push_back(edge);
```

Comment:
This is a bit obscure as coded.  A simple 'if' would be easier for a maintainer to grasp down the road.